zed crash on sway window manager in arch linux when lost focus

### Summary

<!-- Please insert a one line summary of the issue below -->
zed crash on sway window manager in arch linux when lost focus, because zed receive a wired wayland event `Keymap { format: Value(NoKeymap), fd: OwnedFd { fd: 89 }, size: 0 }`, which, cause the following assert
```
impl Dispatch<wl_keyboard::WlKeyboard, ()> for WaylandClientStatePtr {
    fn event(
    ...
    ) {
    ...
                assert_eq!(
                    format,
                    wl_keyboard::KeymapFormat::XkbV1,
                    "Unsupported keymap format"
                );
```
in `crates/gpui/src/platform/linux/wayland/client.rs`, failed.

<!-- Include all steps necessary to reproduce from a clean Zed installation. Be verbose -->
Steps to trigger the problem:
1. start a sway windows manager
2. start zed
3. move focus to other window

Actual Behavior:
crash

Expected Behavior:
not crash

### Zed Version and System Specs

Zed: v0.171.6 (Zed)
OS: Linux Wayland arch unknown
Memory: 62.7 GiB
Architecture: x86_64
GPU: AMD Radeon RX 6600 || AMD open-source driver || 2024.Q4.3 (LLPC)

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2025-02-04T02:07:31.831638694+08:00 [INFO] ========== starting zed ==========
2025-02-04T02:07:31.85279493+08:00 [INFO] Presentation support: "VK_KHR_wayland_surface"
2025-02-04T02:07:31.852883291+08:00 [INFO] Presentation support: "VK_KHR_xcb_surface"
2025-02-04T02:07:31.852954802+08:00 [INFO] Presentation support: "VK_KHR_xlib_surface"
2025-02-04T02:07:31.853026753+08:00 [INFO] Enabling Vulkan Portability
2025-02-04T02:07:31.853091393+08:00 [INFO] Enabling color space support
2025-02-04T02:07:31.863793243+08:00 [INFO] Adapter: "AMD Radeon RX 6600"
2025-02-04T02:07:31.863899324+08:00 [INFO] Ray tracing is supported
2025-02-04T02:07:31.865639249+08:00 [INFO] perform;
2025-02-04T02:07:31.86573128+08:00 [INFO] authenticate;
2025-02-04T02:07:31.865839761+08:00 [INFO] Opening main db
2025-02-04T02:07:31.865913682+08:00 [INFO] read_command;
2025-02-04T02:07:31.866018243+08:00 [INFO] send_secondary_commands;
2025-02-04T02:07:31.866257915+08:00 [INFO] new;
2025-02-04T02:07:31.866333276+08:00 [INFO] socket reader;
2025-02-04T02:07:31.866654529+08:00 [INFO] Opening main db
2025-02-04T02:07:31.868052722+08:00 [INFO] Using git binary path: None
2025-02-04T02:07:31.893371217+08:00 [INFO] No prompt template overrides directory found at /home/algebnaly/.local/share/zed/prompt_overrides. Using built-in prompts.
2025-02-04T02:07:31.89375784+08:00 [INFO] No prompt template overrides directory found at /home/algebnaly/.local/share/zed/prompt_overrides. Using built-in prompts.
2025-02-04T02:07:31.894124094+08:00 [INFO] extensions updated. loading 7, reloading 0, unloading 0
2025-02-04T02:07:31.895738769+08:00 [INFO] activate is not implemented on Linux, ignoring the call
2025-02-04T02:07:31.897381784+08:00 [INFO] perform;
2025-02-04T02:07:31.897456725+08:00 [INFO] authenticate;
2025-02-04T02:07:31.897538595+08:00 [INFO] read_command;
2025-02-04T02:07:31.897829808+08:00 [INFO] socket reader;
2025-02-04T02:07:31.899924228+08:00 [ERROR] DBus error service error org.freedesktop.zbus.Error: The name is not activatable
2025-02-04T02:07:31.900011888+08:00 [WARN] Failed to watch: No such file or directory (os error 2) about ["/home/algebnaly/.config/github-copilot"]
2025-02-04T02:07:31.900339371+08:00 [INFO] Opening main db
2025-02-04T02:07:31.917536701+08:00 [INFO] Using surface present mode MAILBOX
2025-02-04T02:07:31.918111296+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2025-02-04T02:07:31.918290688+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2025-02-04T02:07:31.918382199+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PostMultiplied }
2025-02-04T02:07:31.925408974+08:00 [INFO] Using GPU: GpuSpecs { is_software_emulated: false, device_name: "AMD Radeon RX 6600", driver_name: "AMD open-source driver", driver_info: "2024.Q4.3 (LLPC)" }
2025-02-04T02:07:31.926344313+08:00 [INFO] Using surface present mode MAILBOX
2025-02-04T02:07:31.926433543+08:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-02-04T02:07:31.926990978+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2025-02-04T02:07:31.94116185+08:00 [INFO] Using surface present mode MAILBOX
2025-02-04T02:07:31.941266881+08:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-02-04T02:07:31.946456209+08:00 [INFO] Opening main db
2025-02-04T02:07:31.949199015+08:00 [INFO] activate is not implemented on Linux, ignoring the call
2025-02-04T02:07:31.9541102+08:00 [INFO] set environment variables from shell:/usr/bin/fish, path:/nix/var/nix/profiles/default/bin:/home/algebnaly/.ghcup/bin:/home/algebnaly/.local/bin:/home/algebnaly/.cargo/bin:/home/algebnaly/go/bin:/home/algebnaly/.local/share/pnpm:/home/algebnaly/miniconda3/condabin:/home/algebnaly/.local/share/solana/install/active_release/bin:/home/algebnaly/.nix-profile/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/var/lib/flatpak/exports/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/lib/rustup/bin
2025-02-04T02:07:32.849080512+08:00 [ERROR] {
  "thread": "main",
  "payload": "assertion `left == right` failed: Unsupported keymap format\n  left: NoKeymap\n right: XkbV1",
  "location_data": {
    "file": "crates/gpui/src/platform/linux/wayland/client.rs",
    "line": 1135
  },
  "backtrace": [
    "__libc_start_main+24831993102028"
  ],
  "app_version": "0.171.6",
  "release_channel": "stable",
  "target": "x86_64-unknown-linux-gnu",
  "os_name": "Linux Wayland",
  "os_version": "arch unknown",
  "architecture": "x86_64",
  "panicked_on": 1738606052849,
  "system_id": "f76fd659-aa9a-465f-bf12-65b7d82a67cd",
  "installation_id": "edc4e852-5c63-47b9-aa41-5c14f9d49d20",
  "session_id": "5b631c65-db5a-4847-8f99-8f176d655c65"
}
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/gpui/src/platform/linux/wayland/client.rs
+++ b/crates/gpui/src/platform/linux/wayland/client.rs
@@ -1132,11 +1132,10 @@ impl Dispatch<wl_keyboard::WlKeyboard, ()> for WaylandClientStatePtr {
                 size,
                 ..
             } => {
-                assert_eq!(
-                    format,
-                    wl_keyboard::KeymapFormat::XkbV1,
-                    "Unsupported keymap format"
-                );
+                if format != wl_keyboard::KeymapFormat::XkbV1 {
+                    log::error!("Received keymap format {:?}, expected XkbV1", format);
+                    return;
+                }
                 let xkb_context = xkb::Context::new(xkb::CONTEXT_NO_FLAGS);
                 let keymap = unsafe {
                     xkb::Keymap::new_from_fd(

